config/src/main/java/org/springframework/security/config/annotation/web/configuration/HttpSecurityConfiguration.java

```diff
@@ -0,0 +1,117 @@
+import org.springframework.security.web.context.request.async.WebAsyncManagerIntegrationFilter;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.springframework.security.config.Customizer.withDefaults;
+
+/**
+ * {@link Configuration} that exposes the {@link HttpSecurity} bean.
+ *
+ * @author Eleftheria Stein
+ * @since 5.4
+ */
+@Configuration(proxyBeanMethods = false)
+class HttpSecurityConfiguration {
+	private static final String BEAN_NAME_PREFIX = "org.springframework.security.config.annotation.web.configuration.HttpSecurityConfiguration.";
+	private static final String HTTPSECURITY_BEAN_NAME = BEAN_NAME_PREFIX + "httpSecurity";
+
+	private ObjectPostProcessor<Object> objectPostProcessor;
+
+	private AuthenticationManager authenticationManager;
+
+	private AuthenticationConfiguration authenticationConfiguration;
+
+	private ApplicationContext context;
+
+	@Autowired
+	public void setObjectPostProcessor(ObjectPostProcessor<Object> objectPostProcessor) {
+		this.objectPostProcessor = objectPostProcessor;
+	}
+
+	@Autowired(required = false)
+	void setAuthenticationManager(AuthenticationManager authenticationManager) {
+		this.authenticationManager = authenticationManager;
+	}
+
+	@Autowired
+	public void setAuthenticationConfiguration(
+			AuthenticationConfiguration authenticationConfiguration) {
+		this.authenticationConfiguration = authenticationConfiguration;
+	}
+
+	@Autowired
+	public void setApplicationContext(ApplicationContext context) {
+		this.context = context;
+	}
+
+	@Bean(HTTPSECURITY_BEAN_NAME)
+	@Scope("prototype")
+	public HttpSecurity httpSecurity() throws Exception {
+		WebSecurityConfigurerAdapter.LazyPasswordEncoder passwordEncoder =
+				new WebSecurityConfigurerAdapter.LazyPasswordEncoder(this.context);
+
+		AuthenticationManagerBuilder authenticationBuilder =
+				new WebSecurityConfigurerAdapter.DefaultPasswordEncoderAuthenticationManagerBuilder(this.objectPostProcessor, passwordEncoder);
+		authenticationBuilder.parentAuthenticationManager(authenticationManager());
+
+		HttpSecurity http = new HttpSecurity(objectPostProcessor, authenticationBuilder, createSharedObjects());
+		http
+				.csrf(withDefaults())
+				.addFilter(new WebAsyncManagerIntegrationFilter())
+				.exceptionHandling(withDefaults())
+				.headers(withDefaults())
+				.sessionManagement(withDefaults())
+				.securityContext(withDefaults())
+				.requestCache(withDefaults())
+				.anonymous(withDefaults())
+				.servletApi(withDefaults())
+				.logout(withDefaults())
+				.apply(new DefaultLoginPageConfigurer<>());
+
+		return http;
+	}
+
+	private AuthenticationManager authenticationManager() throws Exception {
+		if (this.authenticationManager != null) {
+			return this.authenticationManager;
+		} else {
+			return this.authenticationConfiguration.getAuthenticationManager();
+		}
+	}
+
+	private Map<Class<?>, Object> createSharedObjects() {
+		Map<Class<?>, Object> sharedObjects = new HashMap<>();
```

Comment:
My understanding is that the `WebSecurityConfigurerAdapter` adds the additional shared objects because they may be overridden within that class, for example using `configure(AuthenticationManagerBuilder auth)`.
When using the `SecurityFilterChain` bean, you would need to expose a `UserDetailsService` bean and I confirmed that the `UserDetailsService` is used even if it's not explicitly added to the shared objects here. 
It was similar for the `ContentNegotiationStrategy` and `AuthenticationTrustResolver`.